src/stcal/resample/resample.py

```diff
@@ -0,0 +1,1122 @@
+                with np.errstate(divide="ignore", invalid="ignore"):
+                    inv_variance = model.var_rnoise**-1
+
+                inv_variance[~np.isfinite(inv_variance)] = 1
+
+                if weight_type != 'ivm':
+                    ny, nx = inv_variance.shape
+
+                    # apply a median filter to smooth the weight at saturated
+                    # (or high read-out noise) single pixels. keep kernel size
+                    # small to still give lower weight to extended CRs, etc.
+                    ksz = weight_type[8 if selective_median else 7 :]
+                    if ksz:
+                        kernel_size = int(ksz)
+                        if not (kernel_size % 2):
+                            raise ValueError(
+                                'Kernel size of the median filter in IVM weighting'
+                                ' must be an odd integer.'
+                            )
+                    else:
+                        kernel_size = 3
+
+                    ivm_copy = inv_variance.copy()
+
+                    if selective_median:
+                        # apply median filter selectively only at
+                        # points of partially saturated sources:
+                        jumps = np.where(
+                            np.logical_and(model.dq & saturation, dqmask)
+                        )
+                        w2 = kernel_size // 2
+                        for r, c in zip(*jumps):
+                            x1 = max(0, c - w2)
+                            x2 = min(nx, c + w2 + 1)
+                            y1 = max(0, r - w2)
+                            y2 = min(ny, r + w2 + 1)
+                            data = ivm_copy[y1:y2, x1:x2][dqmask[y1:y2, x1:x2]]
+                            if data.size:
+                                inv_variance[r, c] = np.median(data)
+                            # else: leave it as is
+
+                    else:
+                        # apply median to the entire inv-var array:
+                        inv_variance = median_filter(
+                            inv_variance,
+                            size=kernel_size
+                        )
+                    bad_dqmask = np.logical_not(dqmask)
+                    inv_variance[bad_dqmask] = ivm_copy[bad_dqmask]
+
+            else:
+                warnings.warn(
+                    "var_rnoise array not available. "
+                    "Setting drizzle weight map to 1",
+                    RuntimeWarning
+                )
+                inv_variance = 1.0
+
+            result = inv_variance * dqmask
+
+        elif weight_type == 'exptime':
+            exptime = model.meta.exposure.exposure_time
+            result = exptime * dqmask
+
+        else:
+            result = np.ones(model.data.shape, dtype=model.data.dtype) * dqmask
+
+        return result.astype(np.float32)
+
+    @abc.abstractmethod
+    def run(self):
+        ...
```

Comment:
Instead of using ellipses, would it be better to raise and exception with a message saying "method not implemented"?

Reply:
by marking it as an `abstractmethod` this is done automatically if you try to call it directly